js/parts-more/Polar.js

```diff
@@ -535,34 +532,75 @@ if (seriesTypes.column) {
      * @private
      */
     wrap(colProto, 'alignDataLabel', function (proceed, point, dataLabel, options, alignTo, isNew) {
-        if (this.chart.polar) {
-            var angle = point.rectPlotX / Math.PI * 180, align, verticalAlign;
-            // Align nicely outside the perimeter of the columns
-            if (options.align === null) {
-                if (angle > 20 && angle < 160) {
-                    align = 'left'; // right hemisphere
-                }
-                else if (angle > 200 && angle < 340) {
-                    align = 'right'; // left hemisphere
+        var chart = this.chart, inside = pick(options.inside, !!this.options.stacking), angle, align, verticalAlign, shapeArgs, labelPos;
+        if (chart.polar) {
+            angle = point.rectPlotX / Math.PI * 180;
+            if (!chart.inverted) {
+                // Align nicely outside the perimeter of the columns
+                if (options.align === null) {
+                    if (angle > 20 && angle < 160) {
+                        align = 'left'; // right hemisphere
+                    }
+                    else if (angle > 200 && angle < 340) {
+                        align = 'right'; // left hemisphere
+                    }
+                    else {
+                        align = 'center'; // top or bottom
+                    }
+                    options.align = align;
```

Comment:
The wrap is really long, consider to extract logic of angles to separated function.

Reply:
I moved the logic that finds the right alignments to new function called findAlignments. It reduces the alignDataLabel wrap a little bit.